src/release.ss

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env gxi-script
+;; -*- Gerbil -*-
+
+(import
+  :gerbil/gambit/ports
+  :std/format :std/srfi/13 :std/srfi/14 :std/sugar :std/pregexp)
+
+;; Where the root to gerbil is.
+(def gerbil-root
+  (path-normalize (path-expand ".." (path-directory (this-source-file)))))
+
+;; Change the directory to the root of gerbil.
+(current-directory gerbil-root)
+
+;; Check that .git is there indeed
+(or (file-exists? ".git") (error "Cannot find .git directory"))
+
+;; Path to the version file
+(def version-path
+  "src/gerbil/runtime/gx-version.scm")
+
+;; Path to the .gitignore file
+(def ignore-path
+  ".gitignore")
+
+;; Read all the contents of a port as a string
+(def (read-all-as-string port)
+  (let ((string (read-line port #f)))
+    (if (eof-object? string) "" string)))
+
+;; Read all the contents of a port as a list of strings, one per line
+(def (read-all-as-lines
+      port
+      separator: (separator #\newline)
+      include-separator?: (include-separator? #f))
+  (let loop ((lines '()))
+    (let ((line (read-line port separator include-separator?)))
+      (if (string? line)
+        (loop (cons line lines))
+        (reverse lines)))))
+
+;; Read the contents of a file into a list of lines
+(def (read-file-lines file settings: (settings '()))
+  (call-with-input-file (append [path: file] settings) read-all-as-lines))
+
+;; Get the output of a process as a string
+(def (process-output-string path . arguments)
+  (call-with-input-process
+   [path: path arguments: arguments
+    show-console: #f stderr-redirection: #f]
+   read-all-as-string))
+
+;; Just run the process on the current console
+(def (process-run path . arguments)
+  (call-with-output-process
+   [path: path arguments: arguments
+    show-console: #f stderr-redirection: #f stdin-redirection: #f stdout-redirection: #f]
+   void))
+
+;; Get the position of the first character satisfying a predicate in a string
+(def (string-first-position pred string)
+  (let ((length (string-length string)))
+    (let loop ((i 0))
+      (cond
+       ((= i length) #f)
+       ((pred (string-ref string i)) i)
+       (else (loop (+ i 1)))))))
+
+;; Get the position of the last character satisfying a predicate in a string
+(def (string-last-position pred string)
+  (let loop ((i (- (string-length string) 1)))
+    (cond
+     ((< i 0) #f)
+     ((pred (string-ref string i)) i)
+     (else (loop (- i 1))))))
+
+;; Function to compare to an closed argument
+(def (comparing-to x) (lambda (y) (equal? x y)))
+
+;; Compute next development version from the current release version
+(def (next-dev-version version)
+  (string-append version "-DEV"))
+
+;; Compute next release version from the current development version
+(def (next-release-version version)
+  (let* ((dash-position (string-first-position (comparing-to #\-) version))
+         (bare-version (if dash-position (substring version 0 dash-position) version))
+         (dot-position (string-last-position (comparing-to #\.) bare-version))
```

Comment:
Similary this is best written as `(string-index-right bare-version #\.)` with `srfi/13`.
Alternatively you can use the builtin `string-rindex`.